src/Compilers/Core/Portable/Generated/Operations.xml.Generated.cs

```diff
@@ -6548,28 +6548,59 @@ internal sealed partial class LazyTranslatedQueryExpression : BaseTranslatedQuer
         protected override IOperation ExpressionImpl => _lazyExpression.Value;
     }
 
-    internal sealed partial class FlowCapture : Operation, IFlowCaptureOperation
+    internal sealed partial class FlowCaptureReference : Operation, IFlowCaptureReferenceOperation
     {
-        public FlowCapture(int id, SyntaxNode syntax, bool isInitialization, ITypeSymbol type, Optional<object> constantValue) :
-            base(OperationKind.FlowCapture, semanticModel: null, syntax, type, constantValue, isImplicit: true)
+        public FlowCaptureReference(int id, SyntaxNode syntax, ITypeSymbol type, Optional<object> constantValue) :
+            base(OperationKind.FlowCaptureReference, semanticModel: null, syntax, type, constantValue, isImplicit: true)
         {
             Id = id;
-            IsInitialization = isInitialization;
         }
 
         public int Id { get; }
-        public bool IsInitialization { get; }
         public override IEnumerable<IOperation> Children
         {
             get
             {
                 return Array.Empty<IOperation>();
             }
         }
+
+        public override void Accept(OperationVisitor visitor)
+        {
+            visitor.VisitFlowCaptureReference(this);
+        }
+
+        public override TResult Accept<TArgument, TResult>(OperationVisitor<TArgument, TResult> visitor, TArgument argument)
+        {
+            return visitor.VisitFlowCaptureReference(this, argument);
+        }
+    }
+
+    internal sealed partial class FlowCapture : Operation, IFlowCaptureOperation
+    {
+        public FlowCapture(int id, SyntaxNode syntax, IOperation value) :
+            base(OperationKind.FlowCapture, semanticModel: null, syntax, type: null, constantValue: default, isImplicit: true)
+        {
+            Debug.Assert(value != null);
+            Id = id;
+            Value = Operation.SetParentOperation(value, this);
+        }
+
+        public int Id { get; }
+        public IOperation Value { get; }
+        public override IEnumerable<IOperation> Children
+        {
+            get
+            {
+                yield return Value;
```

Comment:
I'd recommend `ImmutableArray.Create(Value)` or some variant (`new[] { Value }`) for same reasons discussed in last PR.

Reply:
To examine the value in the debugger ```.ToArray()``` always works, so I do not believe we should worry about that scenario here. 

---
In reply to: [163065261](https://github.com/dotnet/roslyn/pull/24358#discussion_r163065261) [](ancestors = 163065261)

Reply:
The first reason for returning an array is that iterators generate a pile of IL, which seems unnecessary here.
[https://github.com/dotnet/roslyn/pull/24263/commits/8293ea60c07c67bf8539d89a689b166108cd634f#r162142374](https://github.com/dotnet/roslyn/pull/24263/commits/8293ea60c07c67bf8539d89a689b166108cd634f#r162142374)

---
In reply to: [163069198](https://github.com/dotnet/roslyn/pull/24358#discussion_r163069198) [](ancestors = 163069198,163065261)

Reply:
I prefer to avoid an unnecessary allocation instead. 

---
In reply to: [163072164](https://github.com/dotnet/roslyn/pull/24358#discussion_r163072164) [](ancestors = 163072164,163069198,163065261)

Reply:
From the IL produced, it seems that iterators do allocate too.
```IL
    .method public hidebysig 
        instance class [mscorlib]System.Collections.Generic.IEnumerable`1<int32> M () cil managed 
    {
        .custom instance void [mscorlib]System.Runtime.CompilerServices.IteratorStateMachineAttribute::.ctor(class [mscorlib]System.Type) = (
            01 00 0f 50 72 6f 67 72 61 6d 2b 3c 4d 3e 64 5f
            5f 31 00 00
        )
        // Method begins at RVA 0x2052
        // Code size 8 (0x8)
        .maxstack 8

        IL_0000: ldc.i4.s -2
        IL_0002: newobj instance void Program/'<M>d__1'::.ctor(int32)
        IL_0007: ret
    } // end of method Program::M
```

---
In reply to: [163072907](https://github.com/dotnet/roslyn/pull/24358#discussion_r163072907) [](ancestors = 163072907,163072164,163069198,163065261)

Reply:
I didn't say it didn't, but I prefer to not allocate an array. I believe we don't have a policy banning iterators or prescribing how they should be used. I've made a judgment call that a use of an iterator here is Ok. It feels like you are pushing for establishing a policy, if that is the case, it is a subject for an offline conversation and is out of scope of this PR.   

---
In reply to: [163076168](https://github.com/dotnet/roslyn/pull/24358#discussion_r163076168) [](ancestors = 163076168,163072907,163072164,163069198,163065261)

Reply:
I didn't propose a general policy. I'm just trying to understand the pros/cons to make an informed decision.
If there was no disadvantage, I'd agree with you it's just a matter of preference. But in the present case, it seems there is a disadvantage, but I don't see an advantage. 

---
In reply to: [163078080](https://github.com/dotnet/roslyn/pull/24358#discussion_r163078080) [](ancestors = 163078080,163076168,163072907,163072164,163069198,163065261)